src/cmdline.rs

```diff
@@ -59,50 +90,85 @@ pub enum Command {
     },
 }
 
-/// Get the `App` used for argument parsing.
-pub fn get_app<'a, 'b>() -> App<'a, 'b> {
-    App::new(env!("CARGO_PKG_NAME"))
+fn flag_infer_long_and_short(name: &'static str) -> Arg<'static> {
+    flag_infer_long(name).short(name.chars().next().expect("Name needs at least one char"))
+}
+
+fn flag_infer_long(name: &'static str) -> Arg<'static> {
+    Arg::new(name).long(name)
+}
+
+/// Get the [`clap::Command`] used for argument parsing.
+fn get_clap_command() -> clap::Command<'static> {
+    clap::Command::new(env!("CARGO_PKG_NAME"))
         .version(env!("CARGO_PKG_VERSION"))
-        .setting(AppSettings::TrailingVarArg)
+        .trailing_var_arg(true)
+        .max_term_width(110)
```

Comment:
I enabled the `wrap_help` feature so that the help message doesn't use the more vertically spread out spacing, but when enabled nothing stops the help from just trailing on horizontally forever, so I picked 110 as the length that is large enough to keep things compact, but doesn't trail on forever.

Currently only `--stats-format` has a long enough help line to wrap. Here is it without the wrapping (the usage line never seems to wrap)

```
sccache 0.2.16-alpha.0

USAGE:
    sccache [OPTIONS] <--dist-auth|--dist-status|--show-stats|--start-server|--stop-server|--zero-stats|--package-toolchain <EXE> <OUT>|CMD> [--]

ARGS:
    <CMD>...

OPTIONS:
        --dist-auth                        authenticate for distributed compilation
        --dist-status                      show status of the distributed client
    -h, --help                             Print help information
        --package-toolchain <EXE> <OUT>    package toolchain for distributed compilation
    -s, --show-stats                       show cache statistics
        --start-server                     start background server
        --stats-format <FMT>               set output format of statistics [default: text] [possible values: text, json]
        --stop-server                      stop background server
    -V, --version                          Print version information
    -z, --zero-stats                       zero statistics counters

Enabled features:
    S3:        true
    Redis:     true
    Memcached: true
    GCS:       true
    Azure:     true
```

and then with the current wrapping

```
sccache 0.2.16-alpha.0

USAGE:
    sccache [OPTIONS] <--dist-auth|--dist-status|--show-stats|--start-server|--stop-server|--zero-stats|--package-toolchain <EXE> <OUT>|CMD> [--]

ARGS:
    <CMD>...

OPTIONS:
        --dist-auth                        authenticate for distributed compilation
        --dist-status                      show status of the distributed client
    -h, --help                             Print help information
        --package-toolchain <EXE> <OUT>    package toolchain for distributed compilation
    -s, --show-stats                       show cache statistics
        --start-server                     start background server
        --stats-format <FMT>               set output format of statistics [default: text] [possible values:
                                           text, json]
        --stop-server                      stop background server
    -V, --version                          Print version information
    -z, --zero-stats                       zero statistics counters

Enabled features:
    S3:        true
    Redis:     true
    Memcached: true
    GCS:       true
    Azure:     true
```